streams/src/test/java/org/apache/kafka/test/MockProcessorContext.java

```diff
@@ -51,54 +46,66 @@
 import java.util.List;
 import java.util.Map;
 
-public class MockProcessorContext implements InternalProcessorContext, RecordCollector.Supplier {
+public class MockProcessorContext extends AbstractProcessorContext implements RecordCollector.Supplier {
 
+    private final File stateDir;
+    private final Metrics metrics;
     private final Serde<?> keySerde;
     private final Serde<?> valSerde;
     private final RecordCollector.Supplier recordCollectorSupplier;
-    private final File stateDir;
-    private final Metrics metrics;
-    private final StreamsMetrics streamsMetrics;
-    private final ThreadCache cache;
     private final Map<String, StateStore> storeMap = new LinkedHashMap<>();
-
     private final Map<String, StateRestoreCallback> restoreFuncs = new HashMap<>();
 
     private long timestamp = -1L;
-    private RecordContext recordContext;
-    private ProcessorNode currentNode;
+
+    public MockProcessorContext(final File stateDir, final StreamsConfig config) {
```

Comment:
Nit: please use single parameter per line formatting

Reply:
For these three constructors I feel it not necessary, as it only has two parameters.

Reply:
Yes. But if we add some more parameters later on, it would simplify the diff. But it's also ok to keep as it.

Reply:
Sounds good, will change then.